### PR TITLE
Dev reservations microsoft.capacity 2022 11 01

### PR DESCRIPTION
**Pull request by gaoyp830**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by adrianm32 — APPROVED** [review]

**Comment by TimLovellSmith** [comment]
>         "Cancelled",

Problem from old api version: I am under the impression that provisioningState 'canceled' needs to be spelled with just one 'l' (L) in order for ARM to understand it.
@rpmanda @JeffRichter should they fix this up now even though its a breaking change?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1182 in 4aa9531. [](commit_id = 4aa9531fcbe4fc54edf99718f5837ec9e218e90b, deletion_comment = False)

**Comment by sjanamma** [comment]
>         "summary": "Get the regions and skus that are available for RI purchase for the specified Azure subscription.",

Gets the list of region and .......

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:76 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>         "operationId": "GetCatalog",

This is a List call at the Subscription scope to bring all the catalogs available for RI purchase

ListCatalogs_BySubscriptionId

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:77 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>             "$ref": "./examples/GetCatalog.json"

Create an Example for List Catalogs. Response will be array of Catalogs

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:80 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>     "AppliedReservationList": {

Add "x-ms-idenifiers" for array

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:3057 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>           "$ref": "#/definitions/SubscriptionScopeProperties"

Replace boolean/switch properties with better enum
[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)



---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1950 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>     "CatalogsResult": {

x-ms-identifiers need to be added

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1477 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by sjanamma** [comment]
>           "x-ms-client-flatten": true,

x-ms-client-flatten should not be added here. Can be added to a collection within properties.

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:3044 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "x-ms-identifiers": []

they don't have any property thats a unique identifier or key?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1365 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Fully qualified id of the Savings Plan order being purchased"

resource id?? Or its some other full qualifier like an URI?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2444 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "description": "Subscription that will be charged for purchasing Reservation or Savings Plan"

Try to keep docs in plain english style without a lot of capitalizing?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:3250 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by chchen3** [comment]
> > ```
> >       "description": "Fully qualified id of the Savings Plan order being purchased"
> > ```
> 
> resource id?? Or its some other full qualifier like an URI?
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2444 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)

It is something like "/providers/Microsoft.BillingBenefits/savingsPlanOrders/20000000-0000-0000-0000-000000000000"

**Comment by chchen3** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/appliedReservations": {
> > ```
> 
> This is a list call. Please add "x-ms-pageable": { "nextLinkName": "nextLink" },
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:158 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)

The nextLink is only for the deep nested value inside, not for the response object. When I tried to add, I got error that response object doesn't contain nextLink

**Comment by TimLovellSmith** [comment]
>           "x-ms-identifiers": []

Is it not also
          "x-ms-identifiers": [
            "reservationId"
          ]
?

---
In reply to: [1341524386](https://github.com/Azure/azure-rest-api-specs/pull/21565#issuecomment-1341524386) [](http://example.com/codeflow?ancestors=1341524386,1341524386)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1365 in 7f20066. [](commit_id = 7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f, deletion_comment = False)

**Comment by chchen3** [comment]
> > ```
> >       "x-ms-client-flatten": true,
> > ```
> 
> x-ms-flatten is not supposed to be done on the properties to flatten up to top level properties. Applicable across the PR.
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1651 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)

This change would cause change of SDK's property setup for users. So decided to keep as it is

**Comment by TimLovellSmith** [comment]
>           "x-ms-identifiers": []

I'm surprised that its not symmetrical with reservationsToExchange, which has the reservationId unique identifier?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2236 in 22235cb. [](commit_id = 22235cbe50b6ad860b2ed7c7f383ce1fe57c0c1b, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         },

Why does this savingsPlansToPurchase array not have same x-ms-identifiers as the other?


        },
        "savingsPlansToPurchase": {
          "type": "array",
          "description": "Details of the savings plans being purchased",
          "items": {
            "$ref": "#/definitions/SavingsPlanToPurchaseExchange"
          },
          "x-ms-identifiers": [
            "savingsPlanId"
          ]

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2245 in 22235cb. [](commit_id = 22235cbe50b6ad860b2ed7c7f383ce1fe57c0c1b, deletion_comment = False)

**Comment by chchen3** [comment]
> > ```
> >       "$ref": "#/definitions/SubscriptionScopeProperties"
> > ```
> 
> Replace boolean/switch properties with better enum https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1950 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)

This will introduce a type change which is not discussed in the breaking change meeting. keep it as it is for now.

**Comment by TimLovellSmith** [comment]
>         },

Basically just wondering about the API design here...

---
In reply to: [1344679058](https://github.com/Azure/azure-rest-api-specs/pull/21565#issuecomment-1344679058) [](http://example.com/codeflow?ancestors=1344679058)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2245 in 75bd6d3. [](commit_id = 75bd6d3106060937fcfb917f9569d5b9ada12ad2, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "x-ms-identifiers": []

Basically just wondering about the API design here...

---
In reply to: [1344677160](https://github.com/Azure/azure-rest-api-specs/pull/21565#issuecomment-1344677160) [](http://example.com/codeflow?ancestors=1344677160)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2236 in 75bd6d3. [](commit_id = 75bd6d3106060937fcfb917f9569d5b9ada12ad2, deletion_comment = False)

**Comment by chchen3** [comment]
> > ```
> > "CatalogsResult": {
> > ```
> 
> x-ms-identifiers need to be added
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1477 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)

already added for value list

**Comment by chchen3** [comment]
> > ```
> >       "x-ms-identifiers": []
> > ```
> 
> I'm surprised that its not symmetrical with reservationsToExchange, which has the reservationId unique identifier?
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2236 in [22235cb](https://github.com/Azure/azure-rest-api-specs/commit/22235cbe50b6ad860b2ed7c7f383ce1fe57c0c1b). [](commit_id = [22235cb](https://github.com/Azure/azure-rest-api-specs/commit/22235cbe50b6ad860b2ed7c7f383ce1fe57c0c1b), deletion_comment = False)

The ReservationToPurchase/SavingsPlanToPurchase part contains purchase request, which only contains what type of RI user wants to purchase, so no unique identifier will be presented. It only presents for ReservationsToExchange because that is returning a list of existing reservations. Then in the response of exchange, the purchase request has been received and started in the backend, so a unique id will be generated by then

**Comment by sjanamma** [comment]
>               "$ref": "#/definitions/AppliedReservations"

SInce this is a collection Get, need pagination on it.

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:165 in 6bae505. [](commit_id = 6bae505c71e21f14a4f97e59a8f3e660062a6bad, deletion_comment = False)

**Comment by sjanamma** [comment]
> > > ```
> > > "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/appliedReservations": {
> > > ```
> > 
> > 
> > This is a list call. Please add "x-ms-pageable": { "nextLinkName": "nextLink" },
> > Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:158 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)
> 
> The nextLink is only for the deep nested value inside, not for the response object. When I tried to add, I got error that response object doesn't contain nextLink

Here is the guidance on this from Roopesh "If a lot of elements are expected in an array, model it as a nested child type because you can't provide pagination on a property irrespective of how deep it is. The nested type can then have a LIST call with pagination."

**Comment by chchen3** [comment]
> > > > ```
> > > > "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/appliedReservations": {
> > > > ```
> > > 
> > > 
> > > This is a list call. Please add "x-ms-pageable": { "nextLinkName": "nextLink" },
> > > Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:158 in [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f). [](commit_id = [7f20066](https://github.com/Azure/azure-rest-api-specs/commit/7f20066ecdb8dd12aeecaf1452fd97d4ce2ff20f), deletion_comment = False)
> > 
> > 
> > The nextLink is only for the deep nested value inside, not for the response object. When I tried to add, I got error that response object doesn't contain nextLink
> 
> Here is the guidance on this from Roopesh "If a lot of elements are expected in an array, model it as a nested child type because you can't provide pagination on a property irrespective of how deep it is. The nested type can then have a LIST call with pagination."

Currently the pagination is happening in body->properties->reservationsOrderIds due to initial design. The change of design to the API would introduce breaking change, that will be none compatible with previous versions, so will keep it as it is for this PR.

**Comment by TimLovellSmith** [comment]
>             }

In general for Azure apis that return lists, you're recommended not to return object wrappers. Since you're already fixing that for one GET api maybe you can fix it for the POST api also.

[https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:403 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           },

In general for Azure apis that return lists, you're recommended not to return object wrappers, not arrays. Since you're already fixing that for one GET api maybe you can fix it for the POST api also.

[https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:465 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           },

I'm also unclear about why this returns a list. If merging creates a single reservation, why doesn't it just return one ReservationResponse?

---
In reply to: [1347424616](https://github.com/Azure/azure-rest-api-specs/pull/21565#issuecomment-1347424616) [](http://example.com/codeflow?ancestors=1347424616)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:465 in 75bd6d3. [](commit_id = 75bd6d3106060937fcfb917f9569d5b9ada12ad2, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "description": "Moves a `Reservation` to the state it was before archiving.\n",

How bout 'Restores' ?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:673 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "The provisioning sub state of the reservation, e.g. Succeeded",

suggest: `sub-state` ?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2586 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "This is the date-time when the Reservation was purchased."

lowercase reservation in docs

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2598 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Represent the term of Reservation.",

lowercase reservation in docs

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2631 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "put": {

Why is there no "delete" operation?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:273 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "description": "The resource has been created",

resources should return 201 for created

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:302 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>     "ReservationResponse": {

Is this a TrackedResurce or a ProxyResource? Looks like its a tracked resource since it has a 'location'. But it doesn't have a "put" verb. Can you refactor this to 'allOf' inherit one of TrackedResource or ProxyResource from common types, to clarify your intention? (and match your RP's manifest)

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:1893 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "patch": {

Why support "patch" but not "put"?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:571 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "name": {

Why is this inside 'properties' property-bag and not at the top-level scope, same as a ReservationResponse?

Is it not supposed to be same as the reservation's name/key from the resource path?

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2880 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "x-ms-long-running-operation": true,

specify 

        "x-ms-long-running-operation-options": 

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:580 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>     "/providers/Microsoft.Capacity/calculateExchange": {

specify 
        "x-ms-long-running-operation": true,
and
        "x-ms-long-running-operation-options": 

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:872 in 2921c73. [](commit_id = 2921c73fed787e1335302caaf9bea1220a867f8e, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Fully qualified id of the Savings Plan order being purchased"

I wouldn't capitalize 'Savings Plan' in the description text sentences - its a documentation context.

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2492 in d26a64c. [](commit_id = d26a64ccfeb457455ab32d3eaf7020d2149a83af, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "description": "Represent SavingsPlan term in ISO 8601 format.",

savings plan

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2511 in d26a64c. [](commit_id = d26a64ccfeb457455ab32d3eaf7020d2149a83af, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "The Azure Region where the reserved resource lives.",

nit don't capitalize region

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json:2996 in d26a64c. [](commit_id = d26a64ccfeb457455ab32d3eaf7020d2149a83af, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
Signing off or ARM, please follow up with the PR assignee about how to resolve any remaining validation errors.

**Comment by msyyc** [comment]
Hi @gaoyp830 in 2022-03-01, Reservation_AvailableScopes is LRO operation,

https://github.com/Azure/azure-rest-api-specs/blob/fce0b25dda01303f2c70de34031169b5d326998b/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-03-01/reservations.json#L34-L40
But in 2022-11-01, it is not:
https://github.com/Azure/azure-rest-api-specs/blob/f1ef98d76c8d46c2fde8a922b329764f65f3c8a1/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json#L31-L40

From LRO to non-LRO, it is SDK breaking change. Could you tell us the context? Thanks!
cc @tadelesh @qiaozha

**Comment by chchen3** [comment]
> Hi @gaoyp830 in 2022-03-01, Reservation_AvailableScopes is LRO operation,
> 
> https://github.com/Azure/azure-rest-api-specs/blob/fce0b25dda01303f2c70de34031169b5d326998b/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-03-01/reservations.json#L34-L40
> 
> 
> But in 2022-11-01, it is not:
> https://github.com/Azure/azure-rest-api-specs/blob/f1ef98d76c8d46c2fde8a922b329764f65f3c8a1/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json#L31-L40
> 
> From LRO to non-LRO, it is SDK breaking change. Could you tell us the context? Thanks! cc @tadelesh @qiaozha

Hi @msyyc, this is a get API that always return 200. It was marked as LRO in 2019-04-01 version when it was introduced and that was not correct. So we are removing the LRO flag now. The API is used by our UX, so we wouldn't expect a big impact on SDK users

**Comment by msyyc** [comment]
Hi @qiaozha @tadelesh Please help confirm SDK breaking.

**Comment by chchen3** [comment]
> Hi @gaoyp830 in 2022-03-01, Reservation_AvailableScopes is LRO operation,
> 
> https://github.com/Azure/azure-rest-api-specs/blob/fce0b25dda01303f2c70de34031169b5d326998b/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-03-01/reservations.json#L34-L40
> 
> 
> But in 2022-11-01, it is not:
> https://github.com/Azure/azure-rest-api-specs/blob/f1ef98d76c8d46c2fde8a922b329764f65f3c8a1/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json#L31-L40
> 
> From LRO to non-LRO, it is SDK breaking change. Could you tell us the context? Thanks! cc @tadelesh @qiaozha

@msyyc This API have a slow 95 percentile response time, so we decided to keep it as LRO for now for us to add 201/202 response in future versions if needed.

**Comment by gaoyp830** [comment]
> @chchen3 In addtion to LRO change. The operation "GetCatalog" has been changed to pageable operation. Has this breaking change also been approved? Could you please provide some context? Thanks.

@tadelesh this pr already had breaking change approval

**Comment by chchen3** [comment]
> @chchen3 In addtion to LRO change. The operation "GetCatalog" has been changed to pageable operation. Has this breaking change also been approved? Could you please provide some context? Thanks.

@tadelesh Yes, this breaking change is approved by breaking change reviewers. The context for the change is that we are expecting a lot more skus to be returned in the near future. So the current implementation of returning all the skus in one list is limited. We added the paging ability so that when the number of skus is large, we can returned the list in pages. This api also mainly used be by our UX team to show available products for users in portal.

**Review by msyyc — APPROVED** [review]
